cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -236,9 +245,9 @@ private void persistRecordsMessageIds(Deque<MetricValues> metricValues, Map<Topi
     }
   }
 
-  private void persistRecords(Deque<MetricValues> metricValues) throws Exception {
+  private void persistRecords(Queue<MetricValues> metricValues) throws Exception {
     long now = System.currentTimeMillis();
-    long lastRecordTime = metricValues.getLast().getTimestamp();
+    long lastRecordTime = Collections.max(topicLatestTime.values());
```

Comment:
don't do this. You can easily track and find the largest time when inserting into the queue.